bin/test_runner.rb

```diff
@@ -286,10 +292,9 @@ def check_todo_list
   def run
     ####  main running loop
     while true
```

Comment:
As to your point on parallel PR, let's go for efficiency, `while true` is 3 times faster than `loop do`...
```
#!/usr/bin/env ruby
require 'benchmark'
iterations = 15000000
Benchmark.bm do |x|
  x.report('while true') do
    n = 0
    while true do
      break if n >= iterations
      n += 1
    end
  end
  x.report('loop      ') do
    n = 0
    loop do
      break if n >= iterations
      n += 1
    end
  end
end
```

results:
```
       user     system      total        real
while true  0.310000   0.000000   0.310000 (  0.315692)
loop        0.870000   0.000000   0.870000 (  0.871842)
```
So let's stick with `while true`